xarray/tests/test_merge.py

```diff
@@ -329,27 +342,27 @@ def test_merge(self):
         actual = ds2.merge(ds1)
         assert_identical(expected, actual)
 
-        actual = data.merge(data)
+        actual = data.merge(data, compat="no_conflicts")
         assert_identical(data, actual)
-        actual = data.reset_coords(drop=True).merge(data)
+        actual = data.reset_coords(drop=True).merge(data, compat="no_conflicts")
         assert_identical(data, actual)
-        actual = data.merge(data.reset_coords(drop=True))
+        actual = data.merge(data.reset_coords(drop=True), compat="no_conflicts")
         assert_identical(data, actual)
 
-        with pytest.raises(ValueError):
+        with pytest.raises(ValueError, match="conflicting values for variable"):
             ds1.merge(ds2.rename({"var3": "var1"}))
         with pytest.raises(ValueError, match=r"should be coordinates or not"):
-            data.reset_coords().merge(data)
+            data.reset_coords().merge(data, compat="no_conflicts")
         with pytest.raises(ValueError, match=r"should be coordinates or not"):
-            data.merge(data.reset_coords())
+            data.merge(data.reset_coords(), compat="no_conflicts")
 
     def test_merge_broadcast_equals(self):
         ds1 = xr.Dataset({"x": 0})
         ds2 = xr.Dataset({"x": ("y", [0, 0])})
-        actual = ds1.merge(ds2)
+        actual = ds1.merge(ds2, compat="no_conflicts")
         assert_identical(ds2, actual)
 
-        actual = ds2.merge(ds1)
+        actual = ds2.merge(ds1, compat="override")
```

Comment:
ok this is dangerous, the dimensionality of `'x'` in the output depends on the order of merging.

 I intended `compat="override"` to be used when `"x"` is the same in all datasets, not merely 'compatible' as is here. It seems like we would want to at minimum, assert that `ndim` is same for all `x` in all datasets. 
 
Opened https://github.com/pydata/xarray/issues/10094